hasjob/views/index.py

```diff
@@ -861,3 +863,51 @@ def manifest():
 @app.route('/api/1/embed.js', methods=['GET'])
 def embed():
     return app.send_static_file('embed.js')
+
+
+@app.route('/api/1/oembed', methods=['GET'], subdomain='<subdomain>')
+@app.route('/api/1/oembed', methods=['GET'])
+@requestargs('url')
+def oembed(url):
+    """
+    Endpoint to support oEmbed - https://oembed.com/
+
+    Example request - http://hasjob.co/api/1/oembed?url=https://hasjob.co/
+
+    Required for services like embed.ly
+    """
+    oembedjs = {}
+
+    parsed = urlparse(url)
+    if not parsed.netloc.endswith(u'hasjob.co'):
+        # is this a legit hasjob domain or subdomain? if not, then return 404
+        abort(404)
+
+    boardname, _ = parsed.netloc.split('.', 1)
+    if boardname == 'hasjob':
+        # in case the url is https://hasjob.co/?embed=1
+        boardname = 'www'
+
+    board = Board.query.filter_by(name=boardname).first()
+    if not board:
+        abort(404)
+
+    iframeid = 'hasjob-iframe-' + unicode(uuid4())
+
+    if parsed.path in ['/', '']:
```

Comment:
Use the path that comes back from querying the `url_map`. Don't depend on unvalidated client-side input.